### DISCUSSION TITLE
GITHUB_TOKEN rate limit in Github Action

### DISCUSSION THREAD

**Discussion by dougsong** [community/community#63763] [Actions]
### Select Topic Area

Question

### Body

According to this [document](https://docs.github.com/en/rest/overview/resources-in-the-rest-api?apiVersion=2022-11-28#rate-limits-for-requests-from-github-actions):

> When using GITHUB_TOKEN, the rate limit is 1,000 requests per hour per repository. For requests to resources that belong to an enterprise account on GitHub.com, GitHub Enterprise Cloud's rate limit applies, and the limit is 15,000 requests per hour per repository.

But in my repository, the response header that accesses the GITHUB API through GITHUB_TOKEN found that the ratelimit is 60. In the process of using goreleaser, because goreleaser will sleep when the ratelimit is less than 100, it will eventually cause the job to timeout. Why my rate limit is 60, Instead of 1000 as described in the documentation

my job：https://github.com/dougsong/pulumi-volcengine/actions/runs/5863103337/job/15896010917

x-ratelimit-limit: 60
x-ratelimit-remaining: 58
x-ratelimit-reset: 1692071444
x-ratelimit-used: 2
x-ratelimit-resource: core

![image](https://github.com/community/community/assets/39510437/3560732b-921e-4e8f-9614-7ebbafdd7459)

**Comment by vladpereverzyev** [comment]
The `60` limit usually indicates that the request is being treated as unauthenticated.

Since the request is being made from GitHub Actions, I would first verify the rate limit independently using the same workflow environment:

```yaml
- name: Check GitHub API rate limit
  env:
    GH_TOKEN: ${{ github.token }}
  run: |
    gh api rate_limit --jq '.resources.core | {limit, remaining, used, reset}'
```

Then compare this with the rate limit reported by GoReleaser.

If `gh api` reports the expected authenticated Actions limit while GoReleaser reports `60`, then the issue is likely that GoReleaser is not receiving or using the same token.

Make sure the token is explicitly available to the GoReleaser step, for example:

```yaml
env:
  GITHUB_TOKEN: ${{ github.token }}
```

Also verify which environment variable and authentication mechanism the particular GoReleaser version expects.

The important point is that the `x-ratelimit-*` headers describe the rate limit applied to the **specific API request**. Therefore, seeing:

```text
x-ratelimit-limit: 60
```

means that request is currently subject to the 60-request limit; it does not necessarily mean that the GitHub Actions `GITHUB_TOKEN` itself has a 60-request limit.

I would therefore first confirm that the exact requests made by GoReleaser are authenticated with the Actions token before treating this as a GitHub rate-limit discrepancy.
